### PR TITLE
Extend list of ignored packages

### PR DESCRIPTION
**Pull request by bertschneider**
As instrumenting Mockito can cause instrumentation cycles it's now ignored. Instrumenting Byte Buddy itself could cause similar issues, hence, it's now ignored as well.

Lastly, the whole JUnit package is ignored due to other sub-packages that could distract the fuzzer and slow down startup.

### CODE REVIEW

File: src/main/java/com/code_intelligence/jazzer/junit/AgentConfigurator.java [hunk 1]
```diff
@@ -60,6 +60,6 @@ private static void applyCommonConfiguration() {
         String.join(File.pathSeparator, "com.google.testing.junit.**", "com.intellij.**",
             "io.github.classgraph.**", "junit.framework.**", "net.bytebuddy.**",
             "org.apiguardian.**", "org.assertj.core.**", "org.hamcrest.**", "org.junit.**",
-            "org.opentest4j.**", "org.apache.maven.**", "org.gradle.**"));
+            "org.opentest4j.**", "org.mockito.**", "org.apache.maven.**", "org.gradle.**"));
```
Comment (bertschneider) [inline src/main/java/com/code_intelligence/jazzer/junit/AgentConfigurator.java:63]: @fmeum Why not exclude these all the time?

Comment (bertschneider) [reply src/main/java/com/code_intelligence/jazzer/junit/AgentConfigurator.java:63]: Ok, then let's leave it as is for this PR.

File: .github/workflows/run-all-tests.yml [hunk 2]
```diff
@@ -79,10 +79,10 @@ jobs:
         shell: bash
 
       - name: Build
-        run: bazelisk build ${{env.BUILD_BUDDY_CONFIG}} --java_runtime_version=local_jdk_${{ matrix.jdk }} --disk_cache=${{ matrix.cache }} ${{ matrix.bazel_args }} //...
+        run: bazelisk build ${{env.BUILD_BUDDY_CONFIG}} --java_runtime_version=local_jdk_${{ matrix.jdk }} --disk_cache=${{ matrix.cache }} ${{ matrix.bazel_args }} --build_tag_filters="-no-${{ matrix.arch }}-jdk${{ matrix.jdk }},-no-jdk${{ matrix.jdk }}" //...
```
Comment (fmeum) [inline .github/workflows/run-all-tests.yml:82]: Just want to verify that this is required: Do the tests that use this tag actually fail to build or just fail to run?

**Review by fmeum — APPROVED** [review]

File: .github/workflows/run-all-tests.yml (hunk 2 above)
Comment (bertschneider) [reply .github/workflows/run-all-tests.yml:82]: True, it's not required for this test. In #643 I added (or better tried to) a test which only builds with later Java versions. So I thought, while I'm at it, I add `--build_tag_filters` as well. Adding a differently named set of build tags seems too much. WDYT?

Comment (fmeum) [reply .github/workflows/run-all-tests.yml:82]: Agreed, if this is needed elsewhere than it's fine to add here.

**Pull request merged**